src/components/Welcome.js

```diff
@@ -1,16 +1,31 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
 import getToken from '../lib/tokens';
-import { writeToFirestore } from '../lib/firebase';
+import { writeToFirestore, db } from '../lib/firebase';
 
 const Welcome = ({ setToken }) => {
+  const [input, setInput] = useState('');
+
   const setTokenStorage = () => {
     const token = getToken();
     localStorage.setItem('token', token);
     setToken(token);
     writeToFirestore(token);
   };
 
+  function handleSubmit(event) {
+    event.preventDefault();
+
```

Comment:
I would maybe `.trim()` the input...it's pretty easy to accidentally hit an extra space, and then the token is accepted, but could be confusing for the user.